### PR TITLE
Store :mfa in the Task struct

### LINKED ISSUE #11716: Allow metadata to be added to Task.async and friends
If you return a `Task` to be awaited on, then it is impossible to identify later on what created that task. We should support a `:metadata` field in the `Task` struct with an equivalent `Task.async(fun, metadata: ...)` API on all functions that return `%Task{}`.

### LINKED ISSUE #11720: Attach Task information on Task.await and other exits
If you return a Task to be awaited on, then it is impossible to identify later on what created that task. We should get the task current stacktrace and include that as part of the task exit. We should change the Exception module to format such scenarios properly.

### PR DESCRIPTION
**Pull request by josevalim**
This field can be used for reflection purposes.

Closes #11716.
Closes #11720.

### CODE REVIEW

**Comment by josevalim** [comment]
@whatyouhide I ended up going with a different approach than the one we discussed, by simply storing the MFA. It doesn't require user intervention and is also cheap to maintain.

File: lib/elixir/lib/task.ex [hunk 1]
```diff
@@ -233,8 +233,8 @@ defmodule Task do
     * `:owner` - the PID of the process that started the task
 
   """
-  @enforce_keys [:pid, :ref, :owner]
-  defstruct pid: nil, ref: nil, owner: nil
+  @enforce_keys [:pid, :ref, :owner, :mfa]
+  defstruct [:pid, :ref, :owner, :mfa]
```
Comment (whatyouhide) [inline lib/elixir/lib/task.ex:237]: Little trick that @lexmag was a fan of many years ago:

```suggestion
  defstruct @enforce_keys
```

😛

File: lib/elixir/lib/task.ex [hunk 2]
```diff
@@ -434,11 +441,13 @@ defmodule Task do
 
   Similar to `async/1` except the function to be started is
   specified by the given `module`, `function_name`, and `args`.
+  The `module`, `function_name`, and its arity is stored as
```
Comment (whatyouhide) [inline lib/elixir/lib/task.ex:444]: Typo:

```suggestion
  The `module`, `function_name`, and its arity are stored as
```

**Review by whatyouhide — APPROVED** [review]
Yep, looks good 💯

File: lib/elixir/lib/task.ex (hunk 1 above)
Comment (josevalim) [reply lib/elixir/lib/task.ex:237]: Sourcery!

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**